### PR TITLE
fix: Empty "image_size" in sn_params when "image_snapshot = yes"

### PR DESCRIPTION
**Pull request by PaulYuuu**
In qcontainer.py, when creating snapshot images, will get the image size from params. But if the image_size is defined in different variants, the sn image size may smaller than the base image, which is incorrect. The easier way is empty the image_size in sn_params, to make the size of snapshot image align to the base one, as it's just a snapshot.

ID: 1529

### CODE REVIEW

**Comment by PaulYuuu** [comment]
A reproducer is:

1. Install a guest with `image_size = 50G`.
2. Remove the parameter and use the default one(10G in VT) to boot the guest, but using `image_snapshot = yes`

```
qemu-img create -f qcow2 -b /home/kvm_autotest_root/images/rhel920-aarch64-virtio-scsi.qcow2 -F qcow2 /root/avocado/data/avocado-vt/vl_avocado-vt-vm1_image1.qcow2 10G
```

With the fix:
```
qemu-img create -f qcow2 -b /home/kvm_autotest_root/images/rhel920-aarch64-virtio-scsi.qcow2 -F qcow2 /root/avocado/data/avocado-vt/vl_avocado-vt-vm1_image1.qcow2
```

**Comment by PaulYuuu** [comment]
Hello @YongxueHong @MiriamDeng, feel free to review, thanks.

**Comment by MiriamDeng** [comment]
Will provide the test results today. Thank you !

**Comment by MiriamDeng** [comment]
LGTM, the original issue has gone.

File: virttest/qemu_devices/qcontainer.py [hunk 1]
```diff
@@ -2509,7 +2509,9 @@ def images_define_by_params(self, name, image_params, media=None,
             sn = 'vl_%s_%s' % (self.vmname, name)
             sn_params['image_chain'] = "%s %s" % (name, sn)
             sn_params['image_name'] = sn
-            sn_params['image_size'] = image_params['image_size']
+            # Empty the image_size parameter so that the snapshot will align to
+            # the size of base image
+            sn_params['image_size'] = ""
```
[the remark this answers is not in the document]
Comment (YongxueHong) [reply virttest/qemu_devices/qcontainer.py:2514]: Actually, there is an issue with the attribute conflict of the image that the test case uses the existing image as the system image, but this image's attributes don't match the parameters in the test case like this PR's issue background is that we prepared an existing 50G system image for the next test case to run, but the next case still uses the 20G image size(actually the size is changed) as the system image which will result in the conflict with the existing 50G system image and then met the error about doing the external snapshot.

To sum up, I think the root cause is that we did not follow the normal logic workflow, and should keep the data consistent first

My suggestion: Better to update the attribute of the system image of the test case at the same time.

Hi @luckyh @PaulYuuu 
Please let me know your opinions.
Thanks.

[the remark this answers is not in the document]
Comment (PaulYuuu) [reply virttest/qemu_devices/qcontainer.py:2514]: IMO, the temporary snapshot should follow the image's real metadata to create the snapshot image, not from the parameter. Here I used the easier way to let it align with the base image, but the more reasonable solution is to get the base image's metadata, and then create the snapshot based on it.

In `-drive` usage, only `image_snapshot = yes` is needed to create an internal snapshot, but `-blockdev` dropped this option, so if we also need `image_size` to be provided, then this is a breaking change between `-drive` and `-blockdev`.

However, to keep the same usage as previous(-drive), leave `image_size` empty is one way, another one is still create internal snapshot via `qemu-img snapshot -c`, and we delete the snapshot at the end of the test.
```
qemu-img snapshot -l base.qcow2
qemu-img snapshot -c sn1 base.qcow2
qemu-img snapshot -l base.qcow2
Snapshot list:
ID        TAG               VM SIZE                DATE     VM CLOCK     ICOUNT
1         sn1                   0 B 2023-11-28 22:09:04 00:00:00.000          0
qemu-img snapshot -d sn1 base.qcow2
qemu-img snapshot -l base.qcow2
```

refs:
1. https://wiki.qemu.org/Documentation/CreateSnapshot
2. https://access.redhat.com/documentation/en-us/red_hat_enterprise_linux/7/html/virtualization_deployment_and_administration_guide/sect-using_qemu_img-listing_creating_applying_and_deleting_a_snapshot

**Comment by PaulYuuu** [comment]
Add a `FIXME` as most of the attributes for the snapshot should be obtained from the base image.

**Review by YongxueHong — APPROVED** [review]
